yb-voyager/src/srcdb/oracle.go

```diff
@@ -234,6 +234,43 @@ func (ora *Oracle) FilterUnsupportedTables(tableList []*sqlname.SourceName) []*s
 	return tableList
 }
 
+func (ora *Oracle) FilterEmptyTables(tableList []*sqlname.SourceName) []*sqlname.SourceName {
+	nonEmptyTableList := make([]*sqlname.SourceName, 0)
+	for _, tableName := range tableList {
+		// sql query to find out if it is oracle nested table
+		chkNestedTblquery := fmt.Sprintf("SELECT 1 FROM ALL_NESTED_TABLES WHERE OWNER = '%s' AND TABLE_NAME = '%s'", tableName.SchemaName.Unquoted, tableName.ObjectName.MinQuoted)
+		isNestedTable := 0
+		err := ora.db.QueryRow(chkNestedTblquery).Scan(&isNestedTable)
+		if err != nil && err != sql.ErrNoRows {
+			utils.ErrExit("error in query to check if table %v is a nested table: %v", tableName, err)
+		}
+
+		chkEmptyTblquery := fmt.Sprintf("SELECT 0 FROM %s WHERE ROWNUM=1", tableName.ObjectName.MinQuoted)
+		if isNestedTable == 1 {
+			// query to check empty nested oracle tables
+			chkEmptyTblquery = fmt.Sprintf(`SELECT 0 from dba_segments 
+			where owner = '%s' AND segment_name = '%s' AND segment_type = 'NESTED TABLE'`,
+				tableName.SchemaName.Unquoted, tableName.ObjectName.MinQuoted)
+		}
+
+		var empty int
+		err = ora.db.QueryRow(chkEmptyTblquery).Scan(&empty)
```

Comment:
If you define the earlier suggested `isTableEmpty(...)` at least you can factor our common code between mysql and oracle implementations.

Reply:
done